### PR TITLE
Bug 1854935 - Enable extensions process by default on Nightly and debug builds

### PR DESCRIPTION
**Pull request by willdurand**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- ~~[ ] **Tests**: This PR includes thorough tests or an explanation of why it does not~~
- ~~[ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one~~
- ~~[ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features~~

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1854935

### CODE REVIEW

File: fenix/app/nimbus.fml.yaml [hunk 1]
```diff
@@ -444,7 +451,7 @@ types:
       variants:
         feature-ui:
           description: An integer either 0 or 1 indicating if the UI for cookie banner handling should be visible,
-            0 to hide the UI and 1 to show the UI. The actual UI is composed by cookie banner section 
+            0 to hide the UI and 1 to show the UI. The actual UI is composed by cookie banner section
```
Comment (Amejia481) [inline fenix/app/nimbus.fml.yaml:454]: nit: Unrelated change :)

**Review by Amejia481 — APPROVED** [review]
🎉

File: fenix/app/src/androidTest/java/org/mozilla/fenix/extensions/ExtensionProcessTest.kt [hunk 2]
```diff
@@ -53,7 +53,20 @@ class ExtensionProcessTest {
...
+            "extensions-process" to JSONObject(
+                """
+                  {
+                    "enabled": false
+                  }
+                """.trimIndent(),
+            ),
+        )
+
+        hardcodedNimbus.connectWith(FxNimbus)
```
Comment (Amejia481) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/extensions/ExtensionProcessTest.kt:68]: I forgot about this test 😅, thanks for updating it.

**Review by jhugman — COMMENTED** [review]
A small drive by review; Arturo has already approved this, my review doesn't change that.

File: fenix/app/nimbus.fml.yaml [hunk 3]
```diff
@@ -277,6 +277,13 @@ features:
         description: If true, the extensions process is enabled.
         type: Boolean
         default: false
+    defaults:
+      - channel: developer
+        value:
+          enabled: true
+      - channel: nightly
+        value:
+          enabled: true
```
[the remark this answers is not in the document]
Comment (willdurand) [reply fenix/app/nimbus.fml.yaml]: ah yes, very cool. thanks!
